frame/referenda/src/lib.rs

```diff
@@ -579,6 +606,44 @@ pub mod pallet {
 				};
 			Ok(Some(branch.weight::<T, I>()).into())
 		}
+
+		/// Set the metadata to an ongoing referendum.
+		///
+		/// - `origin`: Must be `Signed`, and the creator of the referendum.
+		/// - `index`: The index of the referendum to add metadata for.
+		/// - `hash`: The preimage hash of an on-chain stored preimage.
+		#[pallet::weight(T::WeightInfo::set_metadata())]
+		pub fn set_metadata(
+			origin: OriginFor<T>,
+			index: ReferendumIndex,
+			hash: PreimageHash,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let status = Self::ensure_ongoing(index)?;
+			ensure!(status.submission_deposit.who == who, Error::<T, I>::NoPermission);
+			ensure!(T::Preimages::len(&hash).is_some(), Error::<T, I>::BadMetadata);
+			T::Preimages::request(&hash);
+			MetadataOf::<T, I>::insert(index, hash);
```

Comment:
@shawntabrizi here we let the submitter of a referendum to change and clear the metadata even after the the referendum entered the deciding stage. I would rather not, what do you think?
In Kusama, the lowest duration to enter the state is 30 min. 

I also wonder why the submission deposit is never returned? Looks like its part of design, its not optional (second in tuple) https://github.com/paritytech/substrate/blob/afa1b2412d9bdad5b4d360aee488f8fa142bf15c/frame/referenda/src/types.rs#L224

Reply:
talked to Raul, he is not concerned by possibility to change the metadata after the referendum passes the prep period. 
And we are not bothered by not unreserving the submission deposit because its too small.

Reply:
i agree updating the metadata should not be a huge concern. I dont understand what you mean about deposit not being returned?

if something is reserved, it should be returned at the end, else if it is a fee, it should be burned.

Reply:
The submission deposit can not be returned right now, there is no functionality which unreserves the submission deposit.